File: contest07/p04.cpp
#include <iostream>
#include <vector>

// Para acelerar input/output
std::ios_base::sync_with_stdio(false);
std::cin.tie(nullptr);

int f(int k, const std::vector<int>& A, int l, int r) {
    int ans = 0;

    for (int i = l - 1; i < r; ++i) {
        while (k % A[i] == 0) {
            k /= A[i];
        }
        ans += k;
    }

    return ans;
}

int main() {
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(nullptr);

    int testcases;
    std::cin >> testcases;

    while (testcases--) {
        int n, q;
        std::cin >> n >> q;

        std::vector<int> A(n);
        for (int i = 0; i < n; ++i) {
            std::cin >> A[i];
        }

        for (int query = 0; query < q; ++query) {
            int k, l, r;
            std::cin >> k >> l >> r;
            std::cout << f(k, A, l, r) << '\n';
        }
    }

    return 0;
}

File: contest07/p04.py
def f(k: int, A: list[int], l: int, r: int) -> int:
    ans = 0

    for i in range(l-1, r):
        while k % A[i] == 0:
            k = k / A[i]
        ans = ans + k

    return ans

def main():
    testcases = int(input())

    for _ in range(testcases):
        n, q = map(int, input().split())
        A = list(map(int, input().split()))

        for query in range(q):
            k, l, r = map(int, input().split())
            print(int(f(k, A, l, r)))

main()

